### DISCUSSION TITLE
assert failed when executing with --enable-unique-id option

### DISCUSSION THREAD

**Discussion by fdrui** [libMesh/libmesh#3418] [Q&A] [answered]
Hi everyone !
I am currently testing the examples of libMesh with the purpose to use libMesh as a mesh backend in a finite element code for structure and fluid dynamics.
I would like to use the "unique id" information for each mesh object, thus I use the compilation option `--enable-unique-id`. The compilation is made by spack from the master branch of the libMesh repository, using gcc11 and OpenMPI v4.1.2. The other options I use are  `--enable-parmesh`, `--enable-metis` and `--enable-parmetis`, PETSc, VTK.
When I compile and execute the example subdomains_ex1 on 2 procs, I get the following error:
```
[0] src/mesh/distributed_mesh.C, line 1093, compiled Oct 25 2022 at 13:46:43

[1] src/mesh/distributed_mesh.C, line 1093, compiled Oct 25 2022 at 13:46:43
--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD
with errorcode 1.
```
I suppose this happens when the method `prepare_for_use()` is called.
Has anyone an idea about what I am doing wrong?
Thank you for your help!
F.D.

**Comment by roystgnr** [comment, marked as answer]
I'm afraid you were just bitten by the OpenMPI 4.1.2 bug here:

https://github.com/open-mpi/ompi/issues/10525

If you can downgrade your OpenMPI to 4.1.1 (or if you're feeling daring, upgrade; skimming over https://github.com/open-mpi/ompi/tags I'd guess that 4.1.5rc1 and 5.0.0rc9 have the patch by now) that should fix it.

**Reply by fdrui** [reply]
Now it's working! Thank you very much!
